pkg/bundle/publish/catalog_images.go

```diff
@@ -147,10 +182,57 @@ func (o *Options) rebuildCatalogs(ctx context.Context, dstDir string, filesInArc
 	return refs, nil
 }
 
-func (o *Options) buildCatalogImage(ctx context.Context, ref reference.DockerImageReference, dockerfileDir, dcDir string) error {
+func deleteLayer(old string) (v1.Layer, error) {
+	deleteMap := map[string][]byte{}
+	deleteMap[old] = []byte{}
+	return crane.Layer(deleteMap)
+}
+
+func addLayer(new, targetPath string) (v1.Layer, error) {
+	return layerFromFile(targetPath, new)
+}
 
-	dockerfile := filepath.Join(dockerfileDir, "index.Dockerfile")
+func buildCatalogLayer(ref, targetRef string, layer ...v1.Layer) error {
+	logrus.Debugf("Pulling image %s for processing", ref)
+	img, err := crane.Pull(ref)
+	if err != nil {
+		return err
+	}
+
+	// Add new layers to image
+	newImg, err := mutate.AppendLayers(img, layer...)
+	if err != nil {
+		return err
+	}
 
+	// Update image config
+	cfg, err := newImg.ConfigFile()
+	if err != nil {
+		return err
+	}
+
+	labels := map[string]string{
+		containertools.ConfigsLocationLabel: "/configs",
+	}
+	cfg.Config.Labels = labels
+	cfg.Config.Cmd = []string{"serve", "configs"}
+	cfg.Config.Entrypoint = []string{"/bin/opm"}
+	newerImg, err := mutate.Config(newImg, cfg.Config)
+	if err != nil {
+		return err
+	}
+	tag, err := name.NewTag(targetRef)
+	if err != nil {
+		return err
+	}
+	if err := crane.Push(newerImg, tag.String()); err != nil {
```

Comment:
Is this pushing to the remote registry, overwriting the source image?

Reply:
@estroz This does not overwrite the image on the source on the connected side. The e2e test will validate that because the connected registry is off when this is run during the publishing phase.